awx/main/expect/run.py

```diff
@@ -97,14 +96,8 @@ def run_pexpect(args, cwd, env, logfile,
         # enforce usage of an OrderedDict so that the ordering of elements in
         # `keys()` matches `values()`.
         expect_passwords = collections.OrderedDict(expect_passwords)
-    password_patterns = expect_passwords.keys()
-    password_values = expect_passwords.values()
-
-    # pexpect needs all env vars to be utf-8 encoded strings
-    # https://github.com/pexpect/pexpect/issues/512
-    for k, v in env.items():
-        if isinstance(v, six.text_type):
-            env[k] = v.encode('utf-8')
```

Comment:
are we sure that the values are going to be properly strings and not bytes? Does this change leave us trusting the source of these? Couldn't that still go wrong?

Reply:
Nope, this isn't necessary in python3 because strings are unicode by default.

The underlying bug in pexpect doesn't show up in python3:

```
~ cat hang.py && python hang.py
# -*- coding: utf-8 -*-

import pexpect
pexpect.spawn(
    'sleep', ['1'], cwd='/tmp/', env={'FOO': u'Iñtërnâtiônàlizætiøn'}, ignore_sighup=True,
    encoding='utf-8', echo=False
)
```

(it hangs)

```
~ python3 hang.py && echo $?
0
```

(it doesn't hang)

Reply:
![leadingajarantelopegroundsquirrel-size_restricted](https://user-images.githubusercontent.com/214912/52966934-8dd1b880-3376-11e9-9312-18553ed93f7e.gif)

Nope, I've totally got this wrong.  We still need this hack in py3.  I'm going to get this fixed in pexpect, where it belongs.